### PR TITLE
GUVNOR-3524: Do not expose unsupported features by UI (e.g. scorecards, decision trees)

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-3524

This PR amends the Product assemblies and ```HomeProducer``` classes to setup permissions different to Community. ```HomeProducer``` classes were amended to adjust the "Dev Ops" links when access to "Servers (provisioning)" is denied. There were also a couple of other fixes to inconsistencies in the ```kie-wb``` and ```kie-drools-wb``` assemblies. Tested Community and Product ```kie-wb``` and ```kie-drools-wb```

### CODE REVIEW

**Comment by manstis** [comment]
Part of an emsemble:

- https://github.com/AppFormer/uberfire/pull/859
- https://github.com/kiegroup/kie-wb-common/pull/1211
- https://github.com/kiegroup/drools-wb/pull/641
- https://github.com/kiegroup/kie-wb-distributions/pull/631

**Comment by manstis** [comment]
@paulovmr Please check you're OK with the ```HomeProducer``` changes.

@dgutierr Your eye-balls on the permissions setup is welcome!

@cristianonicolai You might be interested as this includes changes to the "monitoring webapp" - I looked at the resulting WARs and they appeared hunky dory, but thought you'd like the opportunity to have a look before this is merged.

**Comment by cristianonicolai** [comment]
@manstis just a minor comment regarding some license headers using _2016_ instead of _2017_, apart from that everything else looks good to me.

File: kie-wb-parent/kie-wb-distribution-wars/kie-wb-monitoring/src/main/productized/resources/security-policy.properties [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#######################################
+
+default.home=HomePerspective
+
+default.permission.perspective.read=false
+default.permission.perspective.create=false
+default.permission.perspective.delete=false
+default.permission.perspective.update=false
+default.permission.perspective.read.HomePerspective=true
+default.permission.perspective.read.ProvisioningManagementPerspective=false
```
Comment (dgutierr) [inline kie-wb-parent/kie-wb-distribution-wars/kie-wb-monitoring/src/main/productized/resources/security-policy.properties]: Is not this redundant? Notice that "read" is denied above for all perspectives.

**Comment by jomarko** [comment]
@manstis I found issue during review of the PR. In the PL jira, you stated that given assets are prohibited by default. However on my fresh instance of workbench given assets are allowed by default. What more, if I prohibit some of them, I have to increase the role priority to get the permission working. I included in the video also setup of the workbench. Maybe I am missing something. 

https://webmshare.com/jzZK6

File: kie-drools-wb-parent/kie-drools-wb-home-page-community/src/main/java/org/kie/workbench/drools/client/home/HomeProducer.java [hunk 2]
```diff
@@ -57,21 +76,34 @@ public HomeModel get() {
...
+        }
+        devOps.addLink(servers);
 
-        model.addShortcut(design);
-        model.addShortcut(devOps);
+        return devOps;
+    }
 
-        return model;
+    private String getDevOpsDescription(final boolean isDeploymentsAuthorized) {
```
Comment (jomarko) [inline kie-drools-wb-parent/kie-drools-wb-home-page-community/src/main/java/org/kie/workbench/drools/client/home/HomeProducer.java]: I see this method 4 times in the PR, any chance to get it into some util class? Sure, I have not investigated deeper if it is possible. So if not applicable, I am fine.

File: kie-drools-wb-parent/kie-drools-wb-webapp/src/main/resources/security-policy.properties [hunk 3]
```diff
@@ -44,3 +48,10 @@ default.permission.project.release=false
 default.permission.repository.configure=false
 default.permission.planner.available=false
 default.permission.globalpreferences.edit=false
+
+# See https://issues.jboss.org/browse/GUVNOR-3524
+default.permission.perspective.read.ProvisioningManagementPerspective=true
+default.permission.editor.read.GuidedDecisionTreeEditorPresenter=true
+default.permission.editor.read.GuidedScoreCardEditor=true
+default.permission.editor.read.ScoreCardXLSEditor=true
+default.permission.editor.read.BPMNDiagramEditor=true
```
Comment (jomarko) [inline kie-drools-wb-parent/kie-drools-wb-webapp/src/main/resources/security-policy.properties]: @manstis  I think this is related to my comment with video?

**Comment by manstis** [comment]
@jomarko Thanks for review.

The _Community_ WARs have all the features enabled by default. The _Product_ WARs have them disabled. You will need to compile ```kie-wb-distributions``` with the ```-Dproductized``` switch to generate the Product WARs (I saw you used the Community ```kie-drools-wb``` which is not exactly the same as BRMS/Business Central). 

The need to increase ```admin``` permissions priority has been noticed by @karreiro (and myself). @karreiro has submitted PR https://github.com/AppFormer/uberfire/pull/861 to fix the need for this. It is awaiting @dgutierr review (and will be merged with @karreiro other PRs for GUVNOR-3200).

File: kie-drools-wb-parent/kie-drools-wb-webapp/src/main/resources/security-policy.properties (hunk 3 above)
Comment (manstis) [reply kie-drools-wb-parent/kie-drools-wb-webapp/src/main/resources/security-policy.properties]: @jomarko Yes, these are the _Community_ defaults. The _Product_ defaults are [here](https://github.com/kiegroup/kie-wb-distributions/pull/631/files#diff-d1e07265f5d3d91c9e0ec6dc3f7a5b6fR52) and [here](https://github.com/kiegroup/kie-wb-distributions/pull/631/files#diff-c762b0be4f563278c01f22ac07fdab2cR53)

**Comment by manstis** [comment]
@dgutierr @cristianonicolai Updates completed.

**Comment by dgutierr** [comment]
@manstis Nice. I guess you will squash before merge, do you?

**Comment by manstis** [comment]
@dgutierr Yep.. when the PR is merged commits can be squashed.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by paulovmr** [comment]
Jenkins please retest this.

**Pull request merged**